src/sentry_unix_pageallocator.c

```diff
@@ -93,21 +100,24 @@ sentry__page_allocator_alloc(size_t size)
             g_alloc->page_offset = 0;
             g_alloc->current_page = NULL;
         }
-    }
-
-    size_t pages = (size + sizeof(struct page_header) + g_alloc->page_size - 1)
-        / g_alloc->page_size;
-    rv = get_pages(pages);
-
-    if (rv) {
-        g_alloc->page_offset = (g_alloc->page_size
-                                   - (g_alloc->page_size * pages
-                                       - (size + sizeof(struct page_header))))
-            % g_alloc->page_size;
-        g_alloc->current_page = g_alloc->page_offset
-            ? rv + g_alloc->page_size * (pages - 1)
-            : NULL;
-        rv += sizeof(struct page_header);
+    } else {
+        // allocate new pages
+        size_t pages
+            = (size + sizeof(struct page_header) + g_alloc->page_size - 1)
+            / g_alloc->page_size;
+        rv = get_pages(pages);
+
+        if (rv) {
+            g_alloc->page_offset
+                = (g_alloc->page_size
+                      - (g_alloc->page_size * pages
+                          - (size + sizeof(struct page_header))))
+                % g_alloc->page_size;
```

Comment:
This is all quite hard to read, could you maybe introduce some intermediary variables and -- if necessary -- explain what they mean?